src/QtiItems/QtiItemAssetCloudFrontReplacer.php

```diff
@@ -0,0 +1,142 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ *
+ */
+
+declare(strict_types=1);
+
+namespace oat\awsTools\QtiItems;
+
+use League\Flysystem\AwsS3v3\AwsS3Adapter;
+use League\Flysystem\Config;
+use oat\awsTools\AwsClient;
+use oat\awsTools\items\ItemCloudFrontReplacement;
+use oat\oatbox\service\ConfigurableService;
+use oat\taoItems\model\render\ItemAssetsReplacement;
+use oat\taoQtiItem\model\compile\QtiAssetReplacer\QtiItemAssetReplacer;
+use oat\taoQtiItem\model\pack\QtiAssetPacker\PackedAsset;
+
+class QtiItemAssetCloudFrontReplacer extends ConfigurableService implements QtiItemAssetReplacer
+{
+    /**
+     * The host for the external source
+     */
+    const OPTION_HOST = 'host';
+
+    /**
+     * The prefix for the path
+     */
+    const OPTION_PREFIX = 'prefix';
+
+    /**
+     * {@inheritDoc}
+     */
+    public function shouldBeReplacedWithExternal(PackedAsset $packetAsset): bool
+    {
+        return !$this->isExcluded($this->getFilenameFormPacket($packetAsset));
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @throws \Exception
+     */
+    public function replaceToExternalSource(PackedAsset $packetAsset, string $itemId): string
+    {
+        $filename = $this->getFilenameFormPacket($packetAsset);
+
+        $awsClient = $this->getAwsClient();
+        $s3Client = $awsClient->getS3Client();
+        $config = new Config();
+
+        $itemAssetReplacement = $this->getItemAssetsReplacement();
+
+        if (!($itemAssetReplacement instanceof ItemCloudFrontReplacement)) {
+            throw new \Exception('The ItemCloudFrontReplacement service should be configured properly for using CloudFront assets.');
```

Comment:
Maybe Exception is too generic, could it be `LogicException`? or something else more specific? :)

Reply:
oops, agree